rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java

```diff
@@ -1148,43 +1148,56 @@ private SendRequest createSvpFundTransactionSendRequest(BtcTransaction transacti
     }
 
     private void processSvpSpendTransactionUnsigned(Keccak256 rskTxHash, Federation proposedFederation, BtcTransaction svpFundTxSigned) {
-        BtcTransaction svpSpendTransactionUnsigned = createSvpSpendTransaction(svpFundTxSigned, proposedFederation);
+        BtcTransaction svpSpendTransactionUnsigned;
+        try {
+            svpSpendTransactionUnsigned = createSvpSpendTransaction(svpFundTxSigned, proposedFederation);
+        } catch (IllegalStateException e){
+            logger.error("[processSvpSpendTransactionUnsigned] Error creating spend transaction {}", e.getMessage());
+            return;
+        }
         updateSvpSpendTransactionValues(rskTxHash, svpSpendTransactionUnsigned);
 
         Coin amountSentToActiveFed = svpSpendTransactionUnsigned.getOutput(0).getValue();
         logReleaseRequested(rskTxHash, svpSpendTransactionUnsigned, amountSentToActiveFed);
         logPegoutTransactionCreated(svpSpendTransactionUnsigned);
     }
 
-    private BtcTransaction createSvpSpendTransaction(BtcTransaction svpFundTxSigned, Federation proposedFederation) {
+    private BtcTransaction createSvpSpendTransaction(BtcTransaction svpFundTxSigned, Federation proposedFederation) throws IllegalStateException {
         BtcTransaction svpSpendTransaction = new BtcTransaction(networkParameters);
         svpSpendTransaction.setVersion(BTC_TX_VERSION_2);
 
-        addSvpSpendTransactionInputs(svpSpendTransaction, svpFundTxSigned, proposedFederation);
+        Script proposedFederationRedeemScript = proposedFederation.getRedeemScript();
+        TransactionOutput outputToProposedFed = searchForOutput(
+            svpFundTxSigned.getOutputs(),
+            proposedFederation.getP2SHScript()
+        ).orElseThrow(() -> new IllegalStateException("[createSvpSpendTransaction] Output to proposed federation was not found in fund transaction."));
+        svpSpendTransaction.addInput(outputToProposedFed);
+        svpSpendTransaction.getInput(0).setScriptSig(createBaseP2SHInputScriptThatSpendsFromRedeemScript(proposedFederationRedeemScript));
+
+        Script flyoverRedeemScript = getFlyoverRedeemScript(bridgeConstants.getProposedFederationFlyoverPrefix(), proposedFederationRedeemScript);
+        Script flyoverOutputScript = ScriptBuilder.createP2SHOutputScript(flyoverRedeemScript);
+        TransactionOutput outputToFlyoverProposedFed = searchForOutput(
+            svpFundTxSigned.getOutputs(),
+            flyoverOutputScript
+        ).orElseThrow(() -> new IllegalStateException("[createSvpSpendTransaction] Output to proposed federation was not found in fund transaction."));
```

Comment:
```suggestion
        ).orElseThrow(() -> new IllegalStateException("[createSvpSpendTransaction] Output to proposed flyover federation was not found in fund transaction."));
```
To differentiate from the other message perhaps?

Reply:
yes thanks!

Reply:
updated